### DISCUSSION TITLE
while submitting form, it is redirecting, which is not indended.

### DISCUSSION THREAD

**Discussion by the-mgi** [react-hook-form/react-hook-form#10513] [Q&A] [answered]
this is working fine, in one of my components, but not on another place.

this is my submit button in form
```
 <ButtonComponent
      buttonText="Save Changes"
      onClick={() => {}}
      color="primary"
      type="submit"
  />
```

**Comment by the-mgi** [comment, marked as answer]
was enclosing a form, in another form that was the reason for redirect.

**Comment by Moshyfawn** [comment]
This is not enough information for us to debug your issue 😉

One possibility can be that your form element doesn't have `handleSubmit` passed to its `onSubmit`. `handleSubmit` calls `event.preventDefault()` internally which disables the default HTML form redirect behaviour

**Reply by the-mgi** [reply]
handleSubmit was there as well, it's just that my one form was inside another one, and both of them had onSubmit handlers for handleSubmit

taking out one form from the other one, it works fine.
